Makefile.core.mk

```diff
@@ -87,12 +87,19 @@ export ISTIO_BIN=$(GOBIN)
 export ISTIO_OUT:=$(TARGET_OUT)
 export ISTIO_OUT_LINUX:=$(TARGET_OUT_LINUX)
 
-# LOCAL_OUT should include architecture where we are currently running versus the desired.
-# This is used when we need to run a build artifact.
+# LOCAL_OUT should point to architecture where we are currently running versus the desired.
+# This is used when we need to run a build artifact during tests or later as part of another
+# target. If we are running in the Linux build container on non Linux hosts, we add the
+# linux binaries to the build dependencies, BUILD_DEPS, which can be added to other targets
+# that would need the Linux binaries (ex. tests).
+BUILD_DEPS:=
 ifeq ($(IN_BUILD_CONTAINER),1)
-  LOCAL_OUT := $(ISTIO_OUT_LINUX)
+  export LOCAL_OUT := $(ISTIO_OUT_LINUX)
+  ifneq ($(GOOS_LOCAL),"linux")
```

Comment:
We've already had people arguing why we are building linux binaries when they aren't needed. And that is even worse since we now build all binaries instead of one. They went from creating a single Mac binary in their dev environment pretty quickly to creating 34 (17 of each type I think looking at the Makefile). Getting them back to 17 seemed a good step.

Reply:
I understand that is a problem. I simply dislike this implementation :)